### ISSUE TITLE
0. 관련 논문 정리

### ISSUE THREAD

**Issue by p-jiho** [p-jiho/thesis#2]
#### - Combining Time Series and Sentiment Analysis for Stock Market Forecasting
    - 저자 : Hsiao-Chuan Chou, Kandethody M. Ramachandran
    - 배경 : Sentiment analysis를 이용해 수행한 다양한 논문들 존재. 이 논문은 그 여러 모델을 비교분석.
                ANN, ARIMA, KNN, RNN, SVR, CNN, LSTM 등 다양한 모델을 사용한 논문들을 분석
    - 데이터
           - news data는 headline만 Investing.com 에서 수집
           - 6개의 기업 수집 : BAC, BA, XOM, UBER, JNJ, AAPL
           - 기간 : 길면 약 11년, 짧으면 약 6개월(기업마다 다름)
           - historical price : Yahoo! Finance website
    - 데이터 정제
           - tokenization : 소문자 변환, 약어 변환
           - 불용어 처리 : NLTK의 stop-words list 사용, 빈도수를 기초해 추가
           - price data : MinMaxScaler 똔,ㄴ log 변환을 통해 변환
    - 평가 방법 : RMSE, MAE, MAPE 사용
    - 데이터 분석
           - data를 섞지 않고 8:2로 분리
           - 시계열 분석의 경우 window size를 3,4,5,10,15, 20, 30 사용
           - 모델 1 : SVR 모델
           - 모델 2 : unidirectional LSTM. dropout 0.25, learning rate 0.001, adam optimizer, epoch 50, linear activation
           - 모델 3 : bidirectional LSTM. single layer.
           - 모델 4 : CNN. 시간이 지날수록 영향이 적어짐. two convolutional layers, one pooling layer, hidden layer, input layer.
           - 모델 5 : CNN-LSTM. 
                      -> two convolutional layer(32, 64), max-pooling layer, LSTM layer of 100units, output layer of one euron
                      -> two convolutional layer(64, 128), max-pooling layer, LSTM layer of 200units, a hidden layer 32 neurons, output layer of one neuron
           - 모델 6 : Pretrained Word2vec. gensim package, 그 후 RNN 사용
           - 모델 7 : Word Embedding. 수집한 textual data를 이용해 임베딩 모델 생성
           - 모델 8 : BERT. headline 1 X 768 of dimension으로 변환, 그 후 SVR 사용
           - 모델 9 : lexicon-based sentiment analysis와 LSTM 결합. headline을 sentiment score로 변환. window size는 error measure이 결정
           - 모델 10 : news headline을 word embedding model로 처리 후 시계열 분석을 통한 predicted price와 결합
                      -> headline은 BERT로 변환, price는 시계열 모델 사용해 결합, 예측은 SVR 사용
                      -> headline은 a sequence of word vector 사용, 그 후 LSTM으로 변환, price는  시계열 모델 사용해 결합
                      -> headline은 a sequence of word vector 사용, 그 후 LSTM으로 변환, price는 NN 사용해 결합
          - 모델 11 : t-1 시간의 종가, t 시간의 sentiment score을 결합해 예측, 예측은 BERT 언어 모델을 기초, 그 후 CNN-LSTM 적용해 예측
          - 모델 12 : sentiment score과 headline vector과 predictied price의 결합. 
    - 결론 
          - stock price prediction에서 여러 모델의 결합에 중점을 둠. 
          - 간단한 price data의 one dimension과 복잡한 textual data의 high dimension이 결합된 결과는 기대에 도달하지 못함.(모델 6, 모델 8)
          - 시계열 모형은 낮은 error을 발생시킴. 하지만 유용하지 못했음.
          - sentiment scaling system은 감정 사전에선 포함되지 않은 경우가 있었음. 해결책은 임베딩 모델을 응용해 보완하는 것. short-term 에서 사용할 수 있음.
          - news가 없이는 매우 큰 error을 생성했음.
          - 모델 11이 가장 낮은 error을 생성했고, word의 정보를 잘 추출했음. 즉, 감정사전을 통한 sentiment score과 이전 시간의 price와 예측된 현재 price를 이용해 현재 주가를 예측. BERT, CNN-LSTM을 사용해 예측.
    - 내가 생각하는 문제 : 기업이 6개밖에 안됌. news 데이터의 수가 적음. 몇 기업은 기간이 많이 짧아 데이터의 양이 부족해보임.

#### - Stock Market Prediction Analysis by Incorporating Social and News Opinion and Sentiment
    - 저자 : Zhaoxia WANG, Seng-Beng Ho, Zhiping LIN
    - 배경 
           - financial analysis 만으로는 부족해 news와 social media data를 활용해야함. 
           - 데이터를 제대로 활용할 수 있는 방법이 없음. 
           - 구조적이지 않고 비선형적인 데이터이므로 NN을 사용할 것.
    - 관련 지식
          - 분석을 위해 사용된 기법과 tool을 중점으로 지식을 쌓음.
          - NN의 toolbox인 tranlm, traingdx보다 MLP NN이 error가 적다.
          - 적절한 window 수를 사용한다면 the adaptive exponential smoothing 방법보다 NN이 유용하다.(MAPE가 2% 미만)
          - genetic algorithm은 NN의 연결가중치와 feature discretization의 한계를 동시에 최적화
          - fuzzy interference system + NN = ANFIS => 빠르고 조작이 쉬움
          - 어떤 논문은 news 뜬 뒤 20분 후를 예측 , 어떤 논문은 social media data 포함.
          - 문제 : sentiment에 대해 deep analysis가 이루어지지 않음. technical parameter가 충분히 고려되지 않음
          - 해결 : NN은 technical parameter 뿐만 아니라 news articles 도 고려됌
    - 데이터 수집
          - python 사용
          - the stock market price data : daily DJIA data, Yahoo! finance -> Open, High, Low, Close, Volume, Adj Closed
          - news article : New York Times news, NY Times Archive API
          - term : 1/1/2007 ~ 31/12/2016, 총 10년
          - 80%는 train data, 20%는 test data로 활용
    - 데이터 분석
          - python 사용
          - 데이터 전처리 : Business, National, U.S 등 여러 지정된 section name으로 필터링. 나머지 section은 ignore -> 대략 400,000개의 article 수집
          - sentiment analysis 수행 : NLTK, Vader Sentiment Analyzer 사용. 
                                                     mixed score, negative score, neutral score, positive score으로 평가 가능
         - 실행 : trainbr, trainlm, traincgb, traingda 총 4개의 방법 중 best 선택
         - stock market price 만 사용한 결과와 news article을 같이 사용한 결과를 비교
         - 종가 데이터만 사용, negative score와 positive score을 news sentiments data로 사용
    - 결과
         - trainlm 알고리즘을 사용한 경우가 best
         - news가 영향을 주는 적절한 날짜는 7일
         - sentiment effect parameter은 다음 날에는 가장 강하고, 시간이 지날수록 점점 약해지도록 설정
         - window size는 2~10으로 변화를 주며 예측, window size는 size 수 만큼의 연이은 day를 활용해 다음 날을 예측
         - MSE를 기준으로 결정한 경우, sentiment effect parameter은 0.01%, window size는 6이 best
    - 결론
        - stock market과 관련된 news senriments는 예측력을 향상시켜줌
        - NN 방법은 적절한 window size를 선택하는 것이 중요

#### - 인터넷 뉴스 콘텐츠 빅데이터를 활용한 종합주가지수 예측
    - 저자 : 유지돈, 이익선
    - 배경 : 여러 연구들이 감성사전을 구축해 뉴스의 긍정, 부정에 대한 감성 점수를 매겨  상관성을 분석하고 예측하는 방향으로 수행하였음.  반어법이나 의미가 불분명한 단어 등에도 계량화를 해야하는데 그 정확성이 떨어지는 것이 문제.
    - 데이터 수집
          - 2014년 종합주가지수 1% 상승 또는 하락한 20일치의 뉴스 데이터 수집 : 약 224,000건
         - 긍정 및 부정에 의거한 빈도수 처리
         - 주가지수에 관련이 높아 보이는 단어 선별하여 어휘 추출
         - 2015년 단일종목 2% 이상 상승 또는 하락한 15일 치 뉴스 데이터 57,000건 수집
         - 추출된 어휘를 통해 뉴스데이터와 종합주가지수 간의 상관성 입증, 예측모델 제시
    - 텍스트 분석
         - 1단계
             - 불필요한 기호와 조사에 대한 분리작업 실시
             - 종합주가지수에 영향을 주지 않는 어휘들 정제
             - 복합명사는 분리되지 않음
             - 독립적으로 의미가 없는 어휘 또는 문장부호 등은 제거
             - 유사한 의미를 내포하지만 다르게 표현된 어휘 통합
        - 2단계
             - Krkwic를 이용해 빈도 분석 -> 상위 빈도의 어휘 추출, 긍정 또는 부정이 판단되는 어휘 추출
             - 추출된 어휘 명사와 동사로 분류
             - 어휘의 영향력을 알아보기 위해 입력 다중회귀분석 실행
             - 설명력이 높은 변수를 알아보기 위해 단계적 다중회귀분석 실행
    - 데이터 분석
        - 추출된 12개의 명사 변수와 종합주가지수 상관분석
        - 입력다중회귀분석 실행 -> 변수의 단위가 동일하므로 비표준화 계수 사용
        - 단계적 다중회귀분석 실행 -> 종합주가지수에 유의미한 변수만 사용된 모델 생성
        - 총 4개의 모델 생성 -> 명사, 동사 변수 각각 입력 다중회귀분석, 단계적 다중회귀분석 실행
        - 동사 + 명사로 추가 분석 -> 2개의 모델 추가
    - 예측 정확도 비교
        - 실제 값과 예측 값의 오차의 평균값으로 측정
        - 상승, 하락 각 10일 정도를 추려 예측 정확도 비교
    - 결론
        - 모든 변수를 사용한 것보다 영향을 많이 주는 변수를 추리는 것이 오차가 더 적음
        - 동사 모델 > 명사 모델 > 동사+명사 모델로 병수의 영향도가 큼
    - 내가 생각하는 문제 : 주관적인 생각이 들어감. 데이터 수집 기간이 너무 적음. 단어간의 연관성은 고려하지 않아보임.

#### - Analysis of Stock Market using Text Mining and Natural Language Processing
    - 저자 : Sheikh Shaugat Abdullah, Mohammad Saiedur Rahaman, Mohammad Saidur Rahman
    - 배경 : Stock market은 루머, 뉴스 등 영향을 받기 때문에 Technical analysis + Fundamental analysis 로 예측을 해야 함. 특히 장기적인 관점에서 필수.
    - 데이터 수집 
          - textual data는 형태가 다양하므로 다양하게 수집해 categorize 해야함. 동일한 시간에 정보 저장
          - 인증된 곳에서 데이터를 수집해야함.
          - 시간에 따라 변할 수 있으므로 keyword, semantic으로 변환해 데이터 베이스 구성
    - 데이터 분석
          - 키워드를 이용해 필터링
          - Apache Open NLP 사용 : machine learning toolkit
          - 데이터의 키워드를 이용하여 비슷한 news의 내용을 키워드로 비교
    - 결론 : news는 결과의 큰 힌트가 될 수 있음. 이 논문은 새로운 관점으로 바라본 것. textual data의 필요성에 대한 관점을 제시

#### - Stock Prediction : Integrating Text Mining Approach using Real-Time News
